packages/expo/src/dom/dom-hooks.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect, type DependencyList } from 'react';
+
+import type { DOMImperativeFactory } from './dom.types';
+
+/**
+ * A React `useImperativeHandle` like hook for DOM components.
+ *
+ */
+export function useDomImperativeHandle<T extends DOMImperativeFactory>(
```

Comment:
Is this a native or web API? What happens when it's called on the wrong platform.